docs/docs/misc/glossary/call_types.md

```diff
@@ -0,0 +1,177 @@
+:::
+
+### Top-level Unconstrained
+
+Contract functions with the `unconstrained` Noir keyword are a special type of function still under development, and their semantics will likely change in the near future. They are used to perform state queries from an off-chain client, and are never included in any transaction. No guarantees are made on the correctness of the result since they rely exclusively on unconstrained oracle calls.
+
+A reasonable mental model for them is that of a `view` Solidity function that is never called in any transaction, and is only ever invoked via `eth_call`. Note that in these the caller assumes that the node is acting honestly by exectuing the true contract bytecode with correct blockchain state, the same way the Aztec version assumes the oracles are returning legitimate data.
+
+### aztec.js
+
+There are three different ways to execute an Aztec contract function using the `aztec.js` library, with close similarities to their [JSON-RPC counterparts](#json-rpc).
+
+#### `simulate`
+
+This is used to get a result out of an execution, either private or public. It creates no transaction and spends no gas. The mental model is fairly close to that of [`eth_call`](#eth_call), in that it can be used to call any type of function, simulate its execution and get a result out of it. `simulate` is also the only way to run [top-level unconstrained functions](#top-level-unconstrained).
+
+#include_code public_getter /noir-projects/noir-contracts/contracts/auth_contract/src/main.nr rust
+
+#include_code simulate_public_getter yarn-project/end-to-end/src/e2e_auth_contract.test.ts typescript
+
+:::warning
+No correctness is guaranteed on the result of `simulate`! Correct execution is entirely optional and left up to the client that handles this request.
+:::
+
+#### `prove`
+
+This creates and returns a transaction request, which includes proof of correct private execution and side-efects. The request is not broadcast however, and no gas is spent. It is typically used in testing contexts to inspect transaction parameters or check private execution failure.
+
+#include_code local-tx-fails /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+
+#### `send`
+
+This is the same as [`prove`](#prove) except it also broadcasts the transaction and returns a receipt. This is how transactions are sent, getting them to be included in blocks and spending gas. It is similar to [`eth_sendTransaction`](#eth_sendtransaction), except it also performs some work on the user's device, namely the production of the proof for the private part of the transaction.
+
+#include_code send_tx yarn-project/end-to-end/src/e2e_card_game.test.ts typescript
+
+Unlike many Ethereum client libraries, `send` does not first `simulate` the call, so when testing for expected public execution failure use `simulate` as `send` will broadcast a valid transaction that reverts.
```

Comment:
Should we not run a simulation for it as well? Seems like an easy "win" 🤷 think it is already an option to the function

Reply:
Hm I just saw that `.send` apparently does simulate things, which seems to not match my experience. It looks like `prove` also does public simulation, which I did not expect. Does this match with your expectations?